### ISSUE TITLE
NextMajor Sandbox fails with Restoring CRONUS Demo Database The system cannot find the file specified

### ISSUE THREAD

**Issue by DavidRoys** [microsoft/navcontainerhelper#3390]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
Trying to create a new container for next major.

**Scripts used to create container and cause the issue**
```
$containerName = 'inbcnz'
$password = 'P@ssw0rd'
$securePassword = ConvertTo-SecureString -String $password -AsPlainText -Force
$credential = New-Object pscredential 'admin', $securePassword
$auth = 'UserPassword'
$artifactUrl = Get-BcArtifactUrl -type 'Sandbox' -country 'nz' -select NextMajor -accept_insiderEula
$licenseFile = 'c:\users\daroys\downloads\Capgemini NZ BC23 Dev+ISV expires 22-APR-2024 5233712.bclicense'
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -credential $credential `
    -auth $auth `
    -accept_insiderEula `
    -artifactUrl $artifactUrl `
    -includeTestToolkit `
    -includeTestLibrariesOnly `
    -assignPremiumPlan `
    -licenseFile $licenseFile `
    -updateHosts

```

**Full output of scripts**
```
BcContainerHelper is version 6.0.6
BcContainerHelper is running as administrator
HyperV is Disabled
UsePsSession is True
Host is Microsoft Windows 10 Pro - 10.0.19045.4046
Docker Client Version is 20.10.18
Docker Server Version is 20.10.18
Removing Session inbcnz
Removing container inbcnz
Removing entries from hosts
Removing inbcnz from container hosts file
Removing inbcnz-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\inbcnz
Fetching all docker images
Fetching all docker volumes
INFO: Windows 10 21H1/21H2 images are not yet available, using 2004 as these are found to work better than 20H2 on 21H1/21H2
Using image mcr.microsoft.com/businesscentral:10.0.19041.1415
Creating Container inbcnz
Style: sandbox
Multitenant: Yes
Version: 24.0.16410.16998
Platform: 24.0.16775.0
Generic Tag: 1.0.2.13
Container OS Version: 10.0.19041.1415 (2004)
Host OS Version: 10.0.19045.4046 (22H2)
Patching prompt.ps1 due to issue #2891
WARNING: Host OS is Windows 10 21H1 or newer and Container OS is 2004, defaulting to process isolation. If you experience problems, add -isolation hyperv.
Using process isolation
Using locale en-NZ
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Downloading C:\ProgramData\BcContainerHelper\Extensions\inbcnz\my\Prompt.ps1
Downloading using WebClient
Using license file c:\users\daroys\downloads\Capgemini NZ BC23 Dev+ISV expires 22-APR-2024 5233712.bclicense
Patching container to install dotnet 6.0.27
Downloading C:\ProgramData\BcContainerHelper\Extensions\inbcnz\my\dotnet6-win.exe
Downloading using WebClient
Patching container to install dotnet 8.0.2 and PowerShell 7.4.1
Downloading C:\ProgramData\BcContainerHelper\Extensions\inbcnz\my\dotnet-win8.exe
Downloading using WebClient
Downloading C:\ProgramData\BcContainerHelper\Extensions\inbcnz\my\powershell-7.4.1-win-x64.msi
Downloading using WebClient
Files in C:\ProgramData\BcContainerHelper\Extensions\inbcnz\my:
- AdditionalOutput.ps1
- dotnet-win8.exe
- dotnet6-win.exe
- HelperFunctions.ps1
- license.bclicense
- MainLoop.ps1
- powershell-7.4.1-win-x64.msi
- Prompt.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container inbcnz from image mcr.microsoft.com/businesscentral:10.0.19041.1415
970df299cc962eff346da33af031b92b16cb175a23d5a6372700488ebc35de5d
Waiting for container inbcnz to be ready
Installing dotnet 6.0.27
Installing dotnet 8.0.0
Installing PowerShell 7.4.1
Using artifactUrl https://bcinsider.azureedge.net/sandbox/24.0.16410.16998/nz
Using installer from C:\Run\210-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\24.0.16410.16998\platform\ServiceTier\Program Files
c:\dl\sandbox\24.0.16410.16998\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\24.0.16410.16998\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\24.0.16410.16998\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\sandbox\24.0.16410.16998\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\24.0.16410.16998\platform
c:\dl\sandbox\24.0.16410.16998\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\24.0.16410.16998\nz\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\24.0.16410.16998\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\24.0.16410.16998\nz\Extensions
Copying Applications
C:\dl\sandbox\24.0.16410.16998\platform\Applications
Copying Applications.NZ
C:\dl\sandbox\24.0.16410.16998\nz\Applications.NZ
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
The system cannot find the file specified
at <ScriptBlock>, C:\Run\navinstall.ps1: line 206
at <ScriptBlock>, C:\Run\start.ps1: line 242
at <ScriptBlock>, <No file>: line 1Error
Installing dotnet 6.0.27
Installing dotnet 8.0.0
Installing PowerShell 7.4.1
Using artifactUrl https://bcinsider.azureedge.net/sandbox/24.0.16410.16998/nz
Using installer from C:\Run\210-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\24.0.16410.16998\platform\ServiceTier\Program Files
c:\dl\sandbox\24.0.16410.16998\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\24.0.16410.16998\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\24.0.16410.16998\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\sandbox\24.0.16410.16998\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\24.0.16410.16998\platform
c:\dl\sandbox\24.0.16410.16998\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\24.0.16410.16998\nz\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\24.0.16410.16998\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\24.0.16410.16998\nz\Extensions
Copying Applications
C:\dl\sandbox\24.0.16410.16998\platform\Applications
Copying Applications.NZ
C:\dl\sandbox\24.0.16410.16998\nz\Applications.NZ
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
The system cannot find the file specified
at <ScriptBlock>, C:\Run\navinstall.ps1: line 206
at <ScriptBlock>, C:\Run\start.ps1: line 242
at <ScriptBlock>, <No file>: line 1
New-BcContainer Telemetry Correlation Id: 364caf34-f4e4-40a2-b755-0da67b89c002
Initialization of container inbcnz failed
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.6\ContainerHandling\Wait-NavContainerReady.ps1:42 char:17
+ ...             throw "Initialization of container $containerName failed" ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Initialization ...r inbcnz failed:String) [], RuntimeException
    + FullyQualifiedErrorId : Initialization of container inbcnz failed
 
```

**Screenshots**
None.

**Additional context**
Happens every time. Have tried with country nz and us. Have tried regenerating script with latest Wizard. Also fails on another laptop.

**Comment by freddydk** [comment]
Right now, NextMajor requires insider BcContainerHelper - please use latest pre-release.

**Comment by KrunoslavPipic** [comment]
What about sandbox latest? 


```
$ContainerName = Read-Host 'Enter container name'
$Credential = Get-Credential

$artifactUrl = Get-BCArtifactUrl -country w1 -select Latest -storageAccount bcartifacts -type Sandbox
$Params = @{}
$Params += @{ artifactUrl = $artifactUrl }
$Params += @{ containerName = $ContainerName }
$Params += @{ auth = 'NavUserPassword' }
$Params += @{ credential = $Credential }
$Params += @{ isolation = 'process' }
$Params += @{ updateHosts = $true }
$Params += @{ accept_outdated  = $true }
$Params += @{ useBestContainerOS = $true }
New-BcContainer @Params
```


```
New-BcContainer Telemetry Correlation Id: d76e2cb0-5fc7-4334-8c19-5d16ad685fbc
DockerDo : docker: Error response from daemon: hcsshim::PrepareLayer failed in Win32: The system cannot find the path specified. (0x3).
ExitCode: 125
Commandline: docker run --volume "c:\bcartifacts.cache:c:\dl" --label nav= --env isBcSandbox=Y --label version=23.5.16502.16994 --label platform=23.0.16919.0 --label
 country=W1 --env artifactUrl=https://bcartifacts.azureedge.net/sandbox/23.5.16502.16994/w1 --env multitenant=Y --env licenseFile="" --name sandbox --hostname sandbo
x --env auth=NavUserPassword --env username="kpipic" --env ExitOnError=N --env locale=en-US --env databaseServer="" --env databaseInstance="" --volume "C:\ProgramDat
a\BcContainerHelper:C:\ProgramData\BcContainerHelper" --volume "C:\ProgramData\BcContainerHelper\Extensions\sandbox\my:C:\Run\my" --isolation process --restart unles
s-stopped --env filesOnly=False --env enableApiServices=Y --env useSSL=N --volume "c:\windows\system32\drivers\etc:C:\driversetc" --env securePassword=76492d1116743f
0423413b16050a5345MgB8ADUAdQBBADAAUQB6AHoAagBtAGEATABsAEMAZQBIAFYARAA0AGoAbQAwAEEAPQA9AHwANwA4AGUANwBlAGMAMAA3AGUAZAA2ADMAZAAxAGMAOABhADkAZgBjADMAMQBhADEAYQBiADkANAB
lAGYAOQA0ADgAOQA0AGEAYwA5ADkAZQBkAGEANQA5AGUANgA3AGEAMQA3ADYAYwA3ADIAYwBkAGYAOABjAGYAOQAzAGIAMgA= --env passwordKeyFile="c:\run\my\aes.key" --env removePasswordKeyFi
le=Y --env accept_eula=Y --env accept_outdated=Y --detach mcr.microsoft.com/businesscentral:10.0.20348.2322
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\6.0.7\ContainerHandling\New-NavContainer.ps1:1997 char:15
+ ...       if (!(DockerDo -accept_eula -accept_outdated:$accept_outdated - ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,DockerDo
```


![image](https://github.com/microsoft/navcontainerhelper/assets/150338832/087b10df-7c44-4fc8-b719-b2bd86644329)

![image](https://github.com/microsoft/navcontainerhelper/assets/150338832/2eea85d4-873b-401a-879c-67fc018c4c63)

**Comment by freddydk** [comment]
@KrunoslavPipic - that is a very different problem.
The errorcode 125 is typically caused by issues with your docker installation.
When I get these I run the cleanup script from https://freddysblog.com/2018/12/11/clean-up-after-yourself-docker-your-mom-isnt-here/ - haven't had it for a while though.

**Comment by aaaasmile** [comment]
@KrunoslavPipic: I have exact the same issue with Docker 4.28 on windows 10. I solved with: 
1) download and run this script  https://raw.githubusercontent.com/Microsoft/nav-arm-templates/master/CleanupAfterDocker.ps1
2) windows restart
3) Create the container. Here I have used the BcContainerHelper from the navcontainerhelper-master repository.

@freddydk  thank you for the suggestion

**Comment by lukas-pfeifhofer** [comment]
The OnPrem artifacts for v23.5 for me works, but the Sandbox artifacts for v23.5 (country=IT) gives me the following error: 

```
Using artifactUrl https://bcartifacts.azureedge.net/onprem/23.5.16502.16757/it
Using installer from C:\Run\210-new
Installing Business Central: multitenant=False, installOnly=False, filesOnly=False, includeTestToolkit=False, includeTestLibrariesOnly=False, includeTestFrameworkOnly=False, includePerformanceToolkit=False, appArtifactPath=c:\dl\onprem\23.5.16502.16757\it, platformArtifactPath=c:\dl\onprem\23.5.16502.16757\platform, databasePath=c:\dl\onprem\23.5.16502.16757\it\database\Demo Database BC (23-0).bak, licenseFilePath=c:\dl\onprem\23.5.16502.16757\it\database\Cronus.bclicense, rebootContainer=True
Installing from artifacts
Starting Local SQL Server
Service 'SQL Server Browser (SQLBrowser)' cannot be started due to the following error: Cannot start service SQLBrowser on computer '.'.
at <ScriptBlock>, C:\Run\navinstall.ps1: line 76
at <ScriptBlock>, C:\Run\start.ps1: line 243
at <ScriptBlock>, <No file>: line 1
```

When I open the container and try to inspect the services, then there exists no **SQL Server** windows service. It seems that is not installed for these artifacts.

**Comment by freddydk** [comment]
@lukas-pfeifhofer - sorry, but there isn't enough info on that output. You should always provide as much info as possible in these issues.
Please update your BcContainerHelper to the latest version and retry whatever operation lead to this problem.
If it still fails, please create a new issue.

**Comment by freddydk** [comment]
BCContainerHelper 6.0.7 has shipped with a fix for this (with generic images 1.0.2.15 or above)
Also note that generic images 1.0.2.16 for the latest Windows versions has shipped.

**Comment by KrunoslavPipic** [comment]
@lukas-pfeifhofer it seems that is problem in docker. Try not to update to latest update: 

![image](https://github.com/microsoft/navcontainerhelper/assets/150338832/e61dfb24-fd53-490e-b4db-5e6a6ac12e79)

It works for me with latest BCContainerHelper that Freddy suggested.
